### PR TITLE
feat(nodata-soa) For one given zone, return a statically configured SOA for all requests which are not A question

### PR DESCRIPTION
**Pull request by Soulou**

### CODE REVIEW

File: handler.go [hunk 1]
```diff
@@ -91,6 +92,33 @@ func (h *GODNSHandler) do(Net string, w dns.ResponseWriter, req *dns.Msg) {
 	}
 	logger.Info("%s lookup %s", remote, Q.String())
 
+	// We don't care of other type of requests than A, so fail fast returning an
+	// empty DNS response, NOERROR status and the SOA in the Authority Section as
+	// required by RFC https://datatracker.ietf.org/doc/html/rfc2308
+	if len(settings.Hosts.Zone) > 0 && q.Qtype != dns.TypeA && strings.HasSuffix(q.Name, settings.Hosts.Zone) {
```
Comment (johnsudaar) [inline handler.go:98]: Should we send a servfail if the query name doens't have the Zone suffix ?

**Review by johnsudaar — APPROVED** [review]

File: handler.go (hunk 1 above)
Comment (Soulou) [reply handler.go:98]: Instead of forwarding then?

Comment (johnsudaar) [reply handler.go:98]: Yeah

Comment (Soulou) [reply handler.go:98]: To say: "godns is for 1 zone only"?

Comment (johnsudaar) [reply handler.go:98]: Exactly

**Review by johnsudaar — APPROVED** [review]

**Pull request merged**